sound/soc/sof/Kconfig

```diff
@@ -136,6 +126,16 @@ config SND_SOC_SOF_DEBUG
 
 if SND_SOC_SOF_DEBUG
 
+config SND_SOC_SOF_DEBUG_PROBES
+	tristate
+	select SND_SOC_SOF_CLIENT
+	select SND_SOC_COMPRESS
+	help
+	  This option enables the data probing feature that can be used to
+	  gather data directly from specific points of the audio pipeline.
+	  This option is not user-selectable but automagically handled by
+	  'select' statements at a higher level.
+
```

Comment:
does this work?

````
config SND_SOC_SOF_HDA_PROBES
        bool
        select SND_SOC_SOF_DEBUG_PROBES
````

so that's a case of selecting a Kconfig that's conditional, that's broken. 

I would also assert that the location of this option does not matter, because it's selected at a different level.

Reply:
@plbossart, but a randconfig can set it, no?
and if it is enabled without SND_SOC_SOF we will have liner error?

Reply:
I don't think this can be set, because there's no menu. It's hidden in menuconfig, isn't it?

Reply:
Well, debug feature does not necessarily mean that you have to recompile a kernel to debug...

See the DMA traces, we have options to enable them.

In my past, I remember a phone company that required all debug/trace capabilities to be available in production devices :-) Extracting PCM data is very helpful in the field.

Reply:
OK, let's drop this patch.